internal/monitors/kubernetes/cluster/metrics/containers.go

```diff
@@ -0,0 +1,49 @@
+package metrics
+
+import (
+	"strings"
+
+	atypes "github.com/signalfx/signalfx-agent/internal/monitors/types"
+	v1 "k8s.io/api/core/v1"
+)
+
+func dimPropsForContainer(pod *v1.Pod) []*atypes.DimProperties {
+	dimPropsList := make([]*atypes.DimProperties, 0)
+
+	for _, cs := range pod.Status.ContainerStatuses {
+
+		if cs.ContainerID == "" {
+			continue
+		}
+
+		containerProps := make(map[string]string)
+
+		if cs.State.Running != nil {
+			containerProps["container_status"] = "running"
+		}
+
+		if cs.State.Terminated != nil {
+			containerProps["container_status"] = "terminated"
+			containerProps["container_status_reason"] = cs.State.Terminated.Reason
+		}
+
+		if cs.State.Waiting != nil {
+			containerProps["container_status"] = "waiting"
+			containerProps["container_status_reason"] = cs.State.Waiting.Reason
+		}
+
+		if len(containerProps) > 0 {
+			dimProps := &atypes.DimProperties{
+				Dimension: atypes.Dimension{
+					Name:  "container_id",
+					Value: strings.Replace(cs.ContainerID, "docker://", "", 1),
```

Comment:
Pull this logic out to a function so it isn't duplicated between `pods.go` and here.

Reply:
Also please confirm that container_id is fully matching between what kubelet-stats and this derives.

Reply:
Yes, confirmed the properties are getting propagated to MTSes from kubelet-stats monitor and addressed your comment.